design-documents/message-queue/PHPEnqueueLibrary.md

```diff
@@ -0,0 +1,54 @@
+# 7- PHP Enqueue Library 
+
+PHP Enqueue provides JMS style abstraction layer over many brokers as discussed here
+
+[PHP Queue Abstraction - Libraries](EvaluationTechnologies.md#PHP-Queue-Abstraction---Libraries)
+
+[Enqueue Quick Tour](https://github.com/php-enqueue/enqueue-dev/blob/master/docs/quick_tour.md)
+
+
+
+## PHP Enqueue Library Evaluation Table - Details
+
+| Method        | Evaluation | Implementation Readiness                                     |
+| ------------- | ---------- | ------------------------------------------------------------ |
+| dequeue()     | Available  | consumer.receive() <br />or simpleclient.consume(new LimitConsumedMessagesExtension($messageLimit)) |
+| acknowledge() | Available  | consumer.acknowledge(message)<br />or Callback function / processor can return ACK |
+| subscribe()   | Available  | consumer.bindCallback(topic, callback function)<br />simpleclient.bindTopic(topic, callback function)<br />consumer.consume() -- blocking call, runs until interrupted<br />simpleclient.consume() -- blocking call, runs until interrupted |
+| reject()      | Available  | consumer.reject(message)<br />or callback function / processor can return REJECT or REQUEUE |
+| push()        | Available  | simpleclient.sendEvent(topic, message)                       |
+
+<img src="legend_img.png" alt="Legend" width="70%" height="70%" />
+
+
+
+## SimpleClient & Queue Interoperablity
+
+There is a Magento2 module available for Enqueue library, [Magento 2 Enqueue Quick Tour](https://github.com/php-enqueue/enqueue-dev/blob/master/docs/magento2/quick_tour.md), this extension provides easy to configure Magento Admin screen, which setup any supported Queue technology within Enqueue (behind the scenes it add configuration here [Magento 2 Enqueue Config Example](https://github.com/php-enqueue/magento2-enqueue/blob/master/etc/config.xml)). In the end, it makes it straightforward to publish and consume messages in your PHP application.
+
+Magento already implements lot of functionality by itself (for instance keeping track of Connection, Queues and their callbacks, consumer types etc.). We can potentially use only the subset of Enqueue library to interface with the undelying Messaging System in a uniform manner; by providing another QueueInterface implemention for **Enqueue** module.
```

Comment:
I guess its a correct approach. Implementing such library will require a lot of time and deep integration in existed implementation. I would not use existed extension and would go to implementation of the subset of Enqueue library